### PR TITLE
Update to github actions and use test-containers for mock server.

### PR DESCRIPTION
**Pull request by jsuereth**
- Update to use test-containers for mock api server (Common java idiom now)
- Move off CircleCi as part of general agreeement/direction AND because GHA supports test containers better out of the box

Does NOT
- Share common code between trace/metric exporters
- do any further cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,10 +14,12 @@ Provides OpenTelemetry Exporters for Google Cloud Operations.
 This project requires a mock server for Google Cloud APIs.  To build and test, do the following:
 
 ```
-$ source get_mock_server.sh
-$ ./gradlew test -Dmock.server.path=$MOCKSERVER
+$ ./gradlew test
 ```
 
+Note: This project uses [Test Containers](http://testcontainers.org), which requires
+docker to be runnable locally by the current users.  Please verify `docker busybox echo hello` works, and if not configure your local docker before building.
```
Comment (punya) [inline README.md:21]:
```suggestion
docker to be runnable locally by the current users.  Please verify `docker run hello-world` works, and if not configure your local docker before building.
```
Either this, or `docker run -it --rm busybox echo hello` if you prefer terse output.

**Review by punya — APPROVED** [review]

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          fetch-depth: 0
+      - id: setup-java-11
+        name: Setup Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - uses: burrunan/gradle-cache-action@v1.6
+        with:
+          remote-build-cache-proxy-enabled: false
+          arguments: spotlessCheck
```
Comment (aabmass) [inline .github/workflows/pr-build.yml:42]: At the risk of bikeshedding, do we want newlines at the end of files? 😃

File: README.md (hunk 1 above)
Comment (aabmass) [reply README.md:21]: I would +1 adding `--rm`

File: exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java [hunk 3]
```diff
@@ -20,63 +20,49 @@
...
+  private static class CloudOperationsMockContainer
+      extends GenericContainer<CloudOperationsMockContainer> {
+    CloudOperationsMockContainer() {
+      super(
+          new ImageFromDockerfile()
+              .withDockerfileFromBuilder(
+                  builder ->
+                      builder
+                          .from("golang")
+                          .run("go get github.com/googleinterns/cloud-operations-api-mock/cmd")
```
Comment (aabmass) [inline exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java:49]: I think the version of the mock API should be pinned somewhere

File: exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java [hunk 4]
```diff
@@ -54,50 +54,37 @@
...
-      String[] cmdArray =
-          new String[] {System.getProperty("mock.server.path"), "-address", address};
-      ProcessBuilder pb = new ProcessBuilder(cmdArray);
-      pb.redirectErrorStream(true);
-      mockServerProcess = pb.start();
-
-      // Setup the mock trace client.
-      mockCloudTraceClient = new MockCloudTraceClient(LOCALHOST, port);
+  /** A test-container instance that loads the Cloud-Ops-Mock server container. */
+  private static class CloudOperationsMockContainer
```
Comment (aabmass) [inline exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java:62]: Is it a lot of work to re-use this test setup in both tests?

File: .github/workflows/pr-build.yml (hunk 2 above)
Comment (jsuereth) [reply .github/workflows/pr-build.yml:42]: You can configure that in spotless.   Separate PR for that, and I'll give you my tirade on style later :)

File: exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java (hunk 3 above)
Comment (jsuereth) [reply exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java:49]: Agreed.  I'm happy to pin to a release, but which one?  *is* there a release that corresponds to the current binary?

File: exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java (hunk 4 above)
Comment (jsuereth) [reply exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java:62]: Right now, yes.  I'll do that in a follow on Cl as I have to restructure the entire build for non-published components.  I had tried to do it but ran into gradle hell and put that off.

I agree it should be done, not in this PR.

**Comment by jamesward** [comment]
@googlebot I consent

File: exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java (hunk 3 above)
Comment (aabmass) [reply exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java:49]: Ya the releases and tags in that repo should be correct https://github.com/googleinterns/cloud-operations-api-mock/releases. The script had `v2-alpha` in there, hopefully go accepts that

**Review by aabmass — APPROVED** [review]

File: exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java (hunk 3 above)
Comment (jsuereth) [reply exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java:49]: It doesn't.  The tags have to be per-module. I think @punya could likely go tag some stuff for us quickly OR I could suck it up and learn me some go go gadget modules.

**Comment by jsuereth** [comment]
Note: This PR is blocked on CircleCI failures (as we migrate off of it).  Waiting for Admin access to make its way to our team, then will submit.

**Pull request merged**